helper/intersectionsParsing.js

```diff
@@ -0,0 +1,134 @@
+const _ = require('lodash');
+/*
+this function returns an object that denotes an intersection of form:
+{
+  street1: value1,
+  street2: value2
+}
+*/
+module.exports = function parseIntersections(text) {
+    var str1 = '', str2 = '';
+    if(text.trim().length > 1) {
+        //   var words = text.toLowerCase().split(' ');
+        var words = _.words(text.toLowerCase(), /[^ ]+/g);
+
+        // changes 'e15' to 'East 15', etc.
+        words = directionalSanitizer(words);
+        // changes '6' to '6th', etc
+        words = addOrdinality(words);
+        // only treat input as intersection if contains '&' or 'and'
+        const delimiter = _.includes(text, '&') ? '&' : 'and';
+        const delimiterIndex = words.indexOf(delimiter);
+
+        str1 = wordsToSentence(words, 0, delimiterIndex);
+        str2 = wordsToSentence(words, delimiterIndex+1, words.length);
+    } else {
+        throw 'Missing streets in the intersection';
```

Comment:
I'm not super familiar with our API middleware code anymore, but what part of the code would catch this exception? My guess is that it would be pretty common for this code to fail to find a valid intersection in the input, so throwing an exception in those cases doesn't feel right. But again, I don't know this code super well anymore, so maybe its ok.

Reply:
@orangejulius You are right. Since I changed the approach to fallback if the intersection query returns nothing, the exception no longer makes sense.

Reply:
Done!